tests/testproject/testproject/settings.py

```diff
@@ -76,10 +76,14 @@
 # https://docs.djangoproject.com/en/2.2/ref/settings/#databases
 
 DATABASES = {
-	'default': {
-		'ENGINE': 'django.db.backends.postgresql_psycopg2',
-		'NAME': 'relativedelta-testproject',
-	},
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'relativedelta-testproject',
+        'HOST': os.environ.get('PG_HOST', '127.0.0.1'),
+        'PORT': os.environ.get('PG_PORT', 5432),
+        'USER': os.environ.get('PG_USER', 'postgres'),
+        'PASSWORD': os.environ.get('PG_PASSWORD', '')
```

Comment:
Not everybody has postgres on same machine and we should not rely on that. As an example I use a docker instance so accessing via network is more portable

Reply:
I think you could just use `PGHOST` and friends from `libpq` then, see [the manual](https://www.postgresql.org/docs/current/libpq-envars.html). This should work out of the box, without explicitly putting this change in here. My point is that this change actively _disables_ that built-in functionality.

Reply:
But the network config is in line with the official Django config: https://docs.djangoproject.com/en/3.1/ref/settings/#databases
Also I'm not removing any functionality (you can always use 127.0.0.1 if you have a local DB but instead I'm adding one: the capability of using a remote DB

Reply:
Those are overrides in Django. These override whatever is done by libpq at a lower level. If you don't set them, it will use whatever is lower down.

>  instead I'm adding one: the capability of using a remote DB

You can use `PGHOST` for that, it will be used by libpq and connect to the specified host.

Reply:
Also, remember that Django is cross-db and most databases don't have such extensive functionality built into their library.

Reply:
I beg to disagree. These are not overrides but rather a quite conventional way of using postgres from django (at the very least the most common setup with django-postgres as quick googling will show ) but I'm not precious with it so if you have another solution I'm happy to take it.
For reference here is the settings that I was planning to introduce later and that I have tested already:

```
db = os.environ.get('DBENGINE', 'pg')
if db == 'sqlite':
    DATABASES = {
        'default': {
            'ENGINE': 'django.db.backends.sqlite3',
            'NAME': 'relativedelta-test',

        }}
elif db == 'mysql':
    DATABASES = {
        'default': {
            'ENGINE': 'django.db.backends.mysql',
            'NAME': 'relativedelta-test',
            'HOST': os.environ.get('MYSQL_HOST', '127.0.0.1'),
            'PORT': os.environ.get('MYSQL_PORT', 3306),
            'USER': os.environ.get('MYSQL_USER', 'reldelta'),
            'PASSWORD': os.environ.get('MYSQL_PASSWORD', ''),
            'OPTIONS': {
                'isolation_level': 'read uncommitted'
            },
            'TEST': {
                'NAME': 'prova',
                'OPTIONS': {
                    'isolation_level': 'read uncommitted'
                },
            },
            'CHARSET': 'utf8',
            'COLLATION': 'utf8_general_ci'}}
else:
    DATABASES = {
        'default': {
            'ENGINE': 'django.db.backends.postgresql_psycopg2',
            'NAME': 'relativedelta-test',
            'HOST': os.environ.get('PG_HOST', '127.0.0.1'),
            'PORT': os.environ.get('PG_PORT', 5432),
            'USER': os.environ.get('PG_USER', 'postgres'),
            'PASSWORD': os.environ.get('PG_PASSWORD', '')}}
```

The above will allow testing on multiple db using tox (it will pass the environment variable accordingly).
Tox will also take care automatically of dropping and recreating the db for each db engine

Reply:
> Also, remember that Django is cross-db and most databases don't have such extensive functionality built into their library.

I'm confused what you mean by "extensive functionality" . What do you mean?

Reply:
With "extensive functionality" I mean having any library-using client automatically pick up on environment variables to determine the behaviour without requiring custom code in clients like you're adding here.

I don't understand how to make tox pick up on the environment variables though. Like I said, I ran tox like so: `PG_HOST=/var/run/postgresql PG_USER=peter tox` and it didn't use these settings.

Reply:
For tox the vaibales must be exported... see my other comment on tox.ini.

As per the "automatically pick up on environment variables" I'm just following best practices see:
- https://12factor.net/
- Line 2 of "the zen of python": Explicit is better than implicit  (having the variables in the settings

But wait ! I now see the whole confusion! In your original post you said: 
"This means if I have a .pgpass in my homedir or I've supplied the PGHOST/PGPORT/PGUSER environment variables which libpq uses by default, those are overridden by these connection options."
If I change the varibales name we got it isn't it?
I mean PGHOST rather than PG_HOST etc...

Reply:
Yes, exactly! Sorry for being so unclear. Also see my comment on the commit; if you let it fall back to `None` it will use the builtins from Postgres when `.env` is missing or the environment variable isn't supplied.